crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -175,6 +190,50 @@ impl Inner {
     }
 }
 
+impl UpdateTask {
+    /// Single run of the background updating process.
+    async fn single_update(&self, inner: &Inner) {
+        let metrics = Metrics::get();
```

Comment:
This is the same as the freestanding function deleted below, except that the outer while loop is in its own function and we use parameters from self.